### ISSUE TITLE
access to dom node from expression

### ISSUE THREAD

**Issue by andreasplesch** [mavoweb/mavo#514]
I want to access a dom node in an expression to retrieve a property from it which was attached by another framework.
Currently, I can use document.querySelector("[property='thisNode']") within an expression to access the dom node of the expression (although I have trouble with all the quoting).
The expression parser/evaluator must already know the dom node. Is there an easier way to access it ?

**Comment by LeaVerou** [comment]
Hello there!

Expressions are about data, not the DOM. The DOM is typically handled by Mavo. Almost always when you need to access DOM nodes in expressions, you've gone down a suboptimal path and there's usually an easier way to do what you're trying to do.

If you want to provide a few more details about what you're trying to do, I might be able to help more effectively! What do you want to do with that node once you get a reference to it?

**Comment by andreasplesch** [comment]
It has a custom property which can be accessed with a custom function like this:

```
[document.querySelector('#BoxMover').getFieldValue('translation')] or
[document.querySelector('#BoxMover')._x3domNode._vf.translation']
```

I then want to use that value, probably with a mv-if.
I can do all that with js but really wanted to give Mavo a try.

**Comment by LeaVerou** [comment]
Even if you do that, `mv-if` won't work correctly because it has no way of knowing when that custom property changes. Is that value reflected in the DOM at all? If so, you can use `property` and `mv-attribute` to "register" it with Mavo. If not, perhaps you can use X3DOM's methods for monitoring when it updates and setting it in the markup?

> I can do all that with js but really wanted to give Mavo a try.

If it would be easier to do it in raw JS, go for it! You can always try Mavo on a project where it's more appropriate :)

**Comment by andreasplesch** [comment]
Yes, I can monitor with x3dom and use setAttribute to another attribute.

Doing something like that I found something strange: #515
